### PR TITLE
Use faster gzip library

### PR DESCRIPTION
**Pull request by dgageot**
This drop in replacement for the `gzip` library reduces the time needed to build an image. 

Before this patch, my two images are built in 34s.
After this patch, my two images are built in 28s.

Signed-off-by: David Gageot <david.gageot@docker.com>

### CODE REVIEW

**Comment by deitch** [comment]
I am not going to complain about a 17% improvement in build times. I wish we could use [zstd](https://github.com/facebook/zstd), but too many things do not support it (including things that need to read a kernel).

File: src/cmd/linuxkit/initrd/initrd.go [hunk 1]
```diff
@@ -3,13 +3,13 @@ package initrd
...
 	"archive/tar"
 	"bytes"
-	"compress/gzip"
 	"errors"
 	"io"
 	"io/ioutil"
 	"path/filepath"
 	"strings"
 
+	gzip "github.com/klauspost/pgzip"
```
Comment (deitch) [inline src/cmd/linuxkit/initrd/initrd.go:12]: I think a comment here that it is a drop-in 100% compatible replacement and 17% faster than `compress/gzip` would be helpful.

File: src/cmd/linuxkit/moby/build.go [hunk 2]
```diff
@@ -18,6 +17,7 @@ import (
 	"strings"
 
 	"github.com/containerd/containerd/reference"
+	gzip "github.com/klauspost/pgzip"
```
Comment (deitch) [inline src/cmd/linuxkit/moby/build.go:20]: Ditto

File: src/cmd/linuxkit/push_packet.go [hunk 3]
```diff
@@ -11,6 +10,7 @@ import (
 	"path/filepath"
 	"runtime"
 
+	"github.com/klauspost/compress/gzip"
```
Comment (deitch) [inline src/cmd/linuxkit/push_packet.go:13]: Why do we use `klauspots/compress/gzip` here and `klauspost/pgzip` elsewhere?

**Review by deitch — CHANGES_REQUESTED** [review]
Some comments please.

File: src/cmd/linuxkit/push_packet.go (hunk 3 above)
Comment (dgageot) [reply src/cmd/linuxkit/push_packet.go:13]: Thanks, it's a mistake

**Pull request merged**